clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -3083,6 +3083,19 @@ ScalarExprEmitter::VisitUnaryExprOrTypeTraitExpr(
                 E->getTypeOfArgument()->getPointeeType()))
             .getQuantity();
     return llvm::ConstantInt::get(CGF.SizeTy, Alignment);
+  } else if (E->getKind() == UETT_VectorElements) {
+    // For scalable vectors, we don't know the size at compile time. We can use
+    // @llvm.vscale to calculate it at runtime.
+    if (E->getTypeOfArgument()->isSizelessVectorType()) {
+      auto *VecTy = dyn_cast<llvm::ScalableVectorType>(
+          ConvertType(E->getTypeOfArgument()));
+      uint64_t NumUnscaledElements = VecTy->getMinNumElements();
+
+      llvm::Value *VScale =
+          Builder.CreateVScale(llvm::ConstantInt::get(CGF.SizeTy, 1));
+      return Builder.CreateMul(
+          VScale, llvm::ConstantInt::get(CGF.SizeTy, NumUnscaledElements));
```

Comment:
You can use `Builder.CreateElementCount(CGF.SizeTy, VecTy->getElementCount())` or so here. (Incidentally, this works for both fixed and scalable vectors.)

Reply:
Neat, thanks for the pointer. that is a lot nicer. @erichkeane this should fix the comments above, as we now handle both types of vectors here and the code is more clear.